<launch>
  <!--DEFINE ARGUMENTS-->

  <arg name="action_topic" default="/tiago/echo_action_server/task2_recognise" />
  <arg name="kimble_path" default="/home/pal/iri-lab/labrobotica/algorithms/erl_person_classifier/person_faces/kimble/kimble.jpeg"  />
  <arg name="postman_path" default="/home/pal/iri-lab/labrobotica/algorithms/erl_person_classifier/person_faces/postman/postman.jpg" />
  <include file="$(find usb_cam)/launch/usb_cam-test.launch"/>


  <node pkg="task2_recognition_client"
 	type="task2_recognition_client"
  	name="task2_recognition_client"
  	output="screen"
  	ns="/tiago">

    <param name="kimble_image_path" value="$(arg kimble_path)" type="str"/>
    <param name="postman_image_path" value="$(arg postman_path)" type="str"/>

  <!-- tts remappings -->
    <remap from="~/tts_module/tts" to="/tts"/>
    <rosparam file="$(find task2_recognition_client)/config/tts.yaml" command="load" ns="tts_module" />

  <!-- face recognition remappings -->
    <remap from="~/face_recognition/get_person" to="/face_recognition_server/get_person" />
    <remap from="~/face_recognition/store_person" to="/face_recognition_server/store_person_path" />

  <!-- Echo Module remappings -->
    <remap from="~/echo_module/listen" to="$(arg action_topic)"/>

  <!-- shirt detection remappings -->
    <remap from="~/shirt_color_detection_module/shirt_color" to="/shirt_color_detection/shirt_color"/>

  </node>

  <node pkg="person_face_recognition" type="face_recognition_server.py" name="face_recognition_server">
      <remap from ="face_recognition_server/camera_image_raw" to="/usb_cam/image_raw"/>
  </node>
  <node pkg="shirt_color_detection" type="shirt_color_detection" name="shirt_color_detection">
      <remap from ="shirt_color_detection/camera_image_raw" to="/usb_cam/image_raw"/>
  </node>

   <!-- launch dynamic reconfigure -->
   <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" respawn="false" output="screen"/>

</launch>
